packages/blitz/src/middleware.ts

```diff
@@ -1,5 +1,14 @@
 import {IncomingMessage, ServerResponse} from "http"
 import {compose, Ctx, RequestMiddleware, MiddlewareNext, MiddlewareResponse} from "./index-server"
+import {UrlObject} from "url"
+import {NextApiResponse} from "next"
+import {resolveHref} from "next/dist/shared/lib/router/router"
+
+const revalidateFactory =
+  (res: NextApiResponse) =>
+  (url: UrlObject | string, opts?: Parameters<NextApiResponse["revalidate"]>[1]) => {
+    return res.revalidate(resolveHref({} as any, url, false), opts)
```

Comment:
Why is `as any` needed?

Reply:
It expects an instance of next/router but only uses it to resolve `#` paths, so we need to pass something here

Reply:
Can you change it to `as NextRouter`?

`NextRouter` imported as:
```
import {resolveHref, NextRouter} from "next/dist/shared/lib/router/router"
```